### PR TITLE
feat:new audit table 

### PR DESCRIPTION
**Pull request by KOTLAPRA23**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-1640

### Change description ###

Logging issue in Profile sync – separating user ID from calls in two tables

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

**Comment by shashank-rastogi** [comment]
is Pact testing changes also part of this PR? ideally they should be in separate PR.. it just make difficult to review with so many files and changes

**Comment by KOTLAPRA23** [comment]
> is Pact testing changes also part of this PR? ideally they should be in separate PR.. it just make difficult to review with so many files and changes

for pact only 2 files change and remaining all are related to audit logic

**Review by shashank-rastogi — COMMENTED** [review]
But are the pact tests verified by QA? since pact is a separate feature, i would prefer it to be moved through a separate PR in prod

File: src/main/java/uk/gov/hmcts/reform/profilesync/schedular/UserProfileSyncJobScheduler.java [hunk 1]
```diff
@@ -41,56 +41,61 @@
...
-
-        SyncJobConfig syncJobConfig =  syncConfigRepository.findByConfigName("firstsearchquery");
+        LocalDateTime startTime = LocalDateTime.now();
+        SyncJobConfig syncJobConfig =  profileSyncConfigRepository.findByConfigName("firstsearchquery");
 
         String configRun =  syncJobConfig.getConfigRun().trim();
-
-        log.info("{}:: Job needs to be run From Last::hours::", loggingComponentName, configRun);
+        ProfileSyncAudit  syncAudit = new ProfileSyncAudit();
+        log.info("{}:: Job needs to be run From Last::hours::{}" + configRun, loggingComponentName);
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/profilesync/schedular/UserProfileSyncJobScheduler.java:55]: please remove + not needed with log.info

File: src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java [hunk 2]
```diff
@@ -38,85 +38,75 @@
...
-    }
-
-    public Map<String, Boolean> addRule(boolean activeFlag, boolean pendingFlag) {
-        Map<String, Boolean> pendingMapWithRules = new HashMap<>();
-        pendingMapWithRules.put("ACTIVE", activeFlag);
-        pendingMapWithRules.put("PENDING", pendingFlag);
-        return pendingMapWithRules;
+        switch (stringBuilder.toString().toLowerCase()) {
+            case "falsetrue":
+                return IdamStatus.PENDING.name();
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java:104]: not getting why we are using switch case ap<Map<String, Boolean>, IdamStatus> idamStatusMap = new HashMap<Map<String, Boolean>, IdamStatus>();

Map is clean code.

File: src/main/resources/db/migration/V1_4__add_sync_audit_new_tables copy.sql [hunk 3]
```diff
@@ -0,0 +1,30 @@
+
```
Comment (shreedhar-hmcts) [inline src/main/resources/db/migration/V1_4__add_sync_audit_new_tables copy.sql:1]: please change the file name, has space also inn file name

File: src/main/java/uk/gov/hmcts/reform/profilesync/schedular/UserProfileSyncJobScheduler.java (hunk 1 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/profilesync/schedular/UserProfileSyncJobScheduler.java:55]: without that loggers not printing proper in the master . that is the fix for loggers issue

File: src/main/java/uk/gov/hmcts/reform/profilesync/schedular/UserProfileSyncJobScheduler.java [hunk 4]
```diff
@@ -41,56 +41,61 @@
...
 
         if (!executeSearchQueryFrom.equals(configRun)) {
 
             searchQuery = searchQuery + configRun;
 
-            log.info("{}:: searchQuery:: will execute from::DB job run value::", loggingComponentName, searchQuery);
+            log.info("{}:: searchQuery:: will execute from::DB job run value::{}" + searchQuery, loggingComponentName);
```
[the remark this answers is not in the document]
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/profilesync/schedular/UserProfileSyncJobScheduler.java:61]: without that loggers not printing proper in the master . that is the fix for loggers issue

File: src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java (hunk 2 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java:104]: where did u find the map code .. i dint see it.

File: src/main/resources/db/migration/V1_4__add_sync_audit_new_tables copy.sql (hunk 3 above)
Comment (KOTLAPRA23) [reply src/main/resources/db/migration/V1_4__add_sync_audit_new_tables copy.sql:1]: this script already added in the demo env and if i change the name then checksum issue will face.

File: src/main/java/uk/gov/hmcts/reform/profilesync/schedular/UserProfileSyncJobScheduler.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/profilesync/schedular/UserProfileSyncJobScheduler.java:55]: without that loggers not printing proper in the master . that is the fix for loggers issue

**Comment by shashank-rastogi** [comment]
This branch has conflicts and needs to be resolved

File: pacts/rd_profile_sync__idam_api-Idam_api.json [hunk 5]
```diff
@@ -0,0 +1,225 @@
+{
+    "provider": {
+        "name": "Idam_api"
```
[the remark this answers is not in the document]
Comment (KOTLAPRA23) [reply pacts/rd_profile_sync__idam_api-Idam_api.json]: it is required but not now so removed it and just kept blank file

File: src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java (hunk 2 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java:104]: it's very hard to readability and understanding so make it simple

File: src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java [hunk 6]
```diff
@@ -38,85 +38,75 @@
...
+                    user.getId());
 
             if (userProfile.isPresent()) {
-                Map<String, Boolean> status = new HashMap<String, Boolean>();
-                status.put(IdamStatus.ACTIVE.name(), user.isActive());
-                status.put(IdamStatus.PENDING.name(), user.isPending());
+                StringBuilder sb = new StringBuilder();
+                sb.append(user.isActive());
+                sb.append(user.isPending());
                 UserProfile updatedUserProfile = UserProfile.builder()
```
Comment (shashank-rastogi) [inline src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java:57]: is nt this mutually exclusive?  why is this required?

**Comment by KOTLAPRA23** [comment]
closing the PR since modified the name of the flyway script

File: src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java (hunk 6 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java:57]: idam some times returns both flags in the response so to support that added above condition
